draft-ietf-httpbis-connect-tcp.md

```diff
@@ -156,11 +157,71 @@ Clients SHOULD assume that all proxy resources generated by a single template sh
 
 ## Closing Connections
 
-In each HTTP version, any requirements related to closing connections in Classic HTTP CONNECT also apply to "connect-tcp", with the following modifications:
-
-* In HTTP/1.1, endpoints SHOULD close the connection in an error state to indicate receipt of a TCP connection error (e.g., a TCP RST or timeout).  Acceptable error states include sending an incomplete DATA capsule (as defined in {{Section 3.3 of !RFC9297}}), a TLS Error Alert ({{!RFC8446, Section 6.2}}), or a TCP RST (if TLS is not in use).  When a connection is terminated in an error state, the receiving endpoint SHOULD send a TCP RST if the underlying TCP implementation permits it.
-* In HTTP/2 and HTTP/3, senders MAY use an incomplete DATA capsule to indicate a TCP connection error, instead of (or in addition to) the signals defined for TCP connection errors in Classic HTTP CONNECT.  Recipients MUST recognize any incomplete capsule as a TCP connection error.
-* Intermediaries MUST propagate connection shutdown errors, including when translating between different HTTP versions.
+Connection termination is essentially symmetrical for proxies and their clients.  In this section, we use the term "endpoint" to describe an implementation of this specification in either role.
+
+When closing connections, endpoints are subject to the following requirements:
+
+* When an endpoint receives a valid TCP FIN, it MUST send a FINAL_DATA capsule.
+* When an endpoint receives a valid FINAL_DATA capsule, it MUST send a TCP FIN.
+* Each endpoint MUST close its send stream when the corresponding TCP connection reaches the TIME-WAIT or CLOSED state.
+  - If the connection closed gracefully, the endpoint MUST close the send stream gracefully.
+  - Otherwise, the endpoint SHOULD close the send stream abruptly, using a mechanism appropriate to the HTTP version:
+    - HTTP/3: RESET_STREAM with H3_CONNECT_ERROR
+    - HTTP/2: RST_STREAM with CONNECT_ERROR
+    - HTTP/1.1 over TLS: a TLS Error Alert
+    - HTTP/1.1 (insecure): TCP RST.
+* The endpoint SHOULD send a TCP RST when the receive stream is terminated if the TCP subsystem permits it and
+  - the receive stream ended abruptly, or
+  - a FINAL_DATA capsule was not received.
```

Comment:
```suggestion
* When the receive stream terminates abruptly or without a FINAL_DATA capsule received, an endpoint SHOULD send a TCP RST if the TCP subsystem permits it.
```
To me this sentence seems to be broken. It might make sense to change to something like this (with the added bonus of each condition being written in the same way).